src/dependency_specifications.rb

```diff
@@ -0,0 +1,31 @@
+
+# frozen_string_literal: true
+
+module Metalware
+  # Class to generate reused `dependency_hash`s, for use by `Dependency` object
+  # to enforce command dependencies.
+  # XXX Consider moving generating of more `dependency_hash`s here.
+  class DependencySpecifications
+    def initialize(config)
+      @config = config
+    end
+
+    def for_node_in_configured_group(node_name)
+      primary_group = configured_primary_group_for_node(node_name)
+      {
+        repo: ['configure.yaml'],
+        configure: ['domain.yaml', "groups/#{primary_group}.yaml"],
```

Comment:
Don't think we want to include `nodes/#{node}.yaml` - every node needs to be part of a configured group, so we need `groups/#{primary_group}.yaml`, but each node in a group does not need to have `metal configure node` run for it, so most `nodes/#{node}.yaml` files will not be found and the dependencies will fail when they should have passed.